cmd/kube-scheduler/app/server.go

```diff
@@ -227,27 +238,12 @@ func Run(cc schedulerserverconfig.CompletedConfig, stopCh <-chan struct{}) error
 	cc.InformerFactory.WaitForCacheSync(stopCh)
 	controller.WaitForCacheSync("scheduler", stopCh, cc.PodInformer.Informer().HasSynced)
 
-	// Prepare a reusable runCommand function.
-	run := func(ctx context.Context) {
-		sched.Run()
-		<-ctx.Done()
-	}
-
-	ctx, cancel := context.WithCancel(context.TODO()) // TODO once Run() accepts a context, it should be used here
-	defer cancel()
-
-	go func() {
-		select {
-		case <-stopCh:
-			cancel()
-		case <-ctx.Done():
-		}
-	}()
-
 	// If leader election is enabled, runCommand via LeaderElector until done and exit.
 	if cc.LeaderElection != nil {
 		cc.LeaderElection.Callbacks = leaderelection.LeaderCallbacks{
-			OnStartedLeading: run,
+			OnStartedLeading: func(context.Context) {
+				sched.Run()
```

Comment:
what does this mean when the context is closed? Will `leaderElector.Run(ctx)` below return ever if this call does not use the context?

Reply:
```
sched.config.StopEverything
```

It seems it use this chan to synchronize over the provided context?

Reply:
Storing a context or stop chan in a struct is generally not preferred. Passing the context into `Scheduler.Run` would be more idiomatic.